src/components/__tests__/SystemPreferences.spec.js

```diff
@@ -0,0 +1,142 @@
+import { mount } from '@vue/test-utils'
+import SystemPreferences from '../SystemPreferences.vue'
+import merge from 'lodash/merge';
+
+// Uncomment this code when moving to vuejs v3:
+// class ShadowRootShadowClass {
+//   // Because the code's not a browser there is no ShadowRoot
+//   // defined for vue to work on.
+//   // See node_modules/@vue/runtime-dom/dist/runtime-dom.cjs.js:1260 for context
+// }
+
+function safeMerge(baseProps, overrides) {
+  let newProps = {};
+  merge(newProps, baseProps, overrides);
+  return newProps;
+}
+function createWrappedPage(props) {
+  // v3: 'propsData' is just 'props'
+  return mount(SystemPreferences, { propsData: props });
+}
+
+function createWrappedPageGetErrorMessages(props) {
+  const wrapper = createWrappedPage(props);
+  const memoryErrorMessage =  wrapper.find("div#memoryInGBWrapper p.bad-input");
+  const cpuErrorMessage = wrapper.find("div#numCPUWrapper p.bad-input");
+  return [memoryErrorMessage, cpuErrorMessage];
+}
+
+const baseProps = {
+  memoryInGB: '4',
+  numberCPUs: 4,
+  availMemoryInGB: 8,
+  availNumCPUs: 6
+};
+
+describe('SystemPreferences.vue', () => {
+  // Uncomment this code when moving to vuejs v3:
+  // beforeAll(() => {
+  //   global.ShadowRoot = ShadowRootShadowClass;
+  // })
+
+  it("accepts valid data", async () => {
+    const wrapper = createWrappedPage(baseProps);
+
+    expect(wrapper.props().memoryInGB).toBe('4');
+    expect(wrapper.props().numberCPUs).toBe(4);
+    expect(wrapper.props().availMemoryInGB).toBe(8);
+    expect(wrapper.props().availNumCPUs).toBe(6);
+
+    // Don't test against the actual value field. See
+    // https://stackoverflow.com/questions/65710738/why-is-the-value-attribute-not-showing-up-when-i-test-this-vue3-component
+    // for details.
```

Comment:
If we add a ref onto the input:
```html
<!-- in SystemPreferences.vue -->
    <div id="memoryInGBWrapper">
      <LabeledInput
        ref="memoryInGB"
```
Then we can test for it:
```javascript
expect(wrapper.getComponent({ ref: "memoryInGB" }).props("value")).toBe("4");
```

Of course, this still doesn't test that the input is correct; for that we need:
```javascript
expect(wrapper.find("#memoryInGBWrapper input").element.value).toBe("4");
```

Reply:
No longer needed now that we're using sliders.